syntax = "proto3";

package endpoints;

// This config is included for potential future use.
message GetArtifactPartConfigMetadata {
}

message GetArtifactPartConfig {
  map<uint32, GetArtifactPartConfigMetadata> metadatas_by_part_number = 1;
}

message BatchGetArtifactPartRequest {
  map<string, GetArtifactPartConfig> configs_by_artifact_id = 1;
}

message GetArtifactPartResultMetadata {
  optional string download_signed_url = 1;
}

message GetArtifactPartResult {
  map<uint32, GetArtifactPartResultMetadata> metadatas_by_part_number = 1;
}

message BatchGetArtifactPartResponse {
  map<string, GetArtifactPartResult> results_by_artifact_id = 1;
}
